tests/test_dataclasses.py

```diff
@@ -2487,3 +2487,25 @@ class A:
     assert count == 0
     TypeAdapter(A).validate_python({'a': 123})
     assert count == 1
+
+
+def test_can_inherit_stdlib_dataclasses_with_defaults():
+    @dataclasses.dataclass
+    class Base:
+        a: None = None
+
+    class Model(BaseModel, Base):
+        pass
+
+    assert Model().a is None
+
+
+def test_can_inherit_stdlib_dataclasses_with_dataclass_fields():
+    @dataclasses.dataclass
+    class Base:
+        a: int = dataclasses.field(default=5)
+
+    class Model(BaseModel, Base):
+        pass
+
+    assert Model().a == 5
```

Comment:
Can we also add a test for `default_factory`? I'm assuming it doesn't work?

Reply:
@adriangb I added a test that xfails. The internals of stdlib dataclasses does not assign default factory attributes to the class instance.

```python
@dataclasses.dataclass
class Base:
    a: int = dataclasses.field(defualt_factory=lambda: 5)

assert hasattr(Base, 'a')  # this fails
```

Reply:
The inheritance works fine if you provide a value, so for my usecase this isn't an important feature to support.